YvesIraguha/lost-and-found-api
# #171926623 registering lost item

#### What does this PR do?
Registering a lost Item
#### Description of Task to be completed?
- Designing a model for lost Item
- Endpoint for lost Item registration
- Testing our feature with jest
#### How should this be manually tested?
- clone the repe `git clone`
- checkout the repo `git checkout ft-register-lost-item-#171926623`
 - Install all dependencies `npm install`
- *Using Postman make a POST request on `http://localhost:3000/api/v1/item/lost`*
data to be sent
 ```javascript
      {
      documentName: "documentName",
      documentNumber: "documentNumber",
      sector: "Sector",
      district: "District",
      isRewarded: "is rewarded",  //true or false
      price: "Price" // price to be issued if is rewarded
}
```
> You should see a returned document
- *Testing using jest*
> run `npm test`
> you Should see a sum of 9 test Suits and 21 tests passing
#### Any background context you want to provide?
- to Send the request in postman you have first to login and set headers to `auth-token` and *Token value* as its value;
- price is required if is rewarded is set to true
#### What are the relevant pivotal tracker stories?
#171926623
